### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by govindappaarun**
Part of this context
 - group contexts
 - add capability to change theme
 - fix components
 - authentication and persist user details & token  in localStorage

### CODE REVIEW

File: src/app-router.js [hunk 1]
```diff
@@ -32,7 +32,11 @@ const privateRoutes = [
 ];
 
 const AppRouter = () => {
-  const appRoutes = publicRoutes.concat(isAuthenticated ? privateRoutes : []);
+  const { authState } = useAuth();
+  console.log({ authState });
```
Comment (samyukthasudhakar) [inline src/app-router.js:36]: console.log statements can be removed.

**Review by samyukthasudhakar — APPROVED** [review]
The code looks great. Neatly structured and highly detailed. Good to merge !
 
Although it would have been helpful if a hosted link of the branch and code files to be reviewed were mentioned as part of the PR.

File: src/pages/header/Header.tsx [hunk 2]
```diff
@@ -35,41 +46,67 @@ export default function Header({ ...rest }: HeaderProps) {
...
+                  {state.quantity}
+                </Badge>
+              </Box>
+              <Box
+                className="icon-container"
+                display="inline-block"
+                onClick={() => navigate("/wishlist")}
+              >
+                <Badge color="success" className="badge">
+                  {state.wishList ? state.wishList.length : 0}
```
Comment (vrinda-mahajan) [inline src/pages/header/Header.tsx:67]: Suggestion: You can use desturcture state here.

File: src/pages/product-list/Components/Product.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import Rating from "src/components/Rating";
+import { Product as ProductType } from "../types";
+import { CardProps } from "src/components/Card/Card";
+import Button, { IconButton } from "src/components/Button";
+import { WishIcon, WishIconSolid, ProductCard } from "../ProductList.styled";
+import { useCart } from "src/contexts";
```
Comment (vrinda-mahajan) [inline src/pages/product-list/Components/Product.tsx:7]: Appreciation: good that you are using absolute path here.

**Review by vrinda-mahajan — APPROVED** [review]
The code looks clean and UI is looking awesome. Good to merge👍

**Pull request merged**